### PR TITLE
[CSDM-730] Compensate chromatic aberration of angular spectrum data

### PR DESCRIPTION
**Pull request by pieleric**
The angular spectrum raw data is typically not straight because of
chromatic aberration. That means the angular values actually (slightly)
varies depending the on the wavelength.
The aberration is already measured at alignment, and stored in the metadata.
However, so far the GUI did nothing with this information. The angles
displayed corresponded to the angles at the center wavelength.

This adds support for compensation, by adjusting the data for each
wavelength independently. This is done just once when loading the data
(and every time a background or coefficient compensation file is
loaded).
This also indirectly means that the 2D view angle vs wavelength is now
linear on the angle axis (instead of being some sort of logarithmic
scale). That makes the graph easier to read.

### CODE REVIEW

File: src/odemis/acq/stream/_static.py [hunk 1]
```diff
@@ -837,8 +838,19 @@ def _updateCalibratedData(self, data=None, bckg=None, coef=None):
             self.calibrated.value = None
             return
 
-        # If MD_THETA_LIST, the length of the A dimension might be reduced
         calibrated = calibration.apply_spectrum_corrections(data, bckg, coef)
+
+        # If angular spectrum, the length of the A dimension might have changed
+        if hasattr(self, "selected_angle"):  # update the list of angles
+            theta_list, _ = spectrum.get_angle_range(calibrated)
+            # Update the range, as it it might has slightly changed
```
Comment (nandishjpatel) [inline src/odemis/acq/stream/_static.py:846]:
```suggestion
            # Update the range, as it might have slightly changed
```

**Review by nandishjpatel — APPROVED** [review]

File: src/odemis/util/angleres.py [hunk 2]
```diff
@@ -619,25 +627,66 @@ def ExtractThetaList(data):
...
+    and the list of angles is derived.
+
+    :param wl > 0: The wavelength, in m, at which these theta values are computed.
+    :param line_bottom: px, px/mm, cf MD_AR_MIRROR_BOTTOM
+    :param line_top: px, px/mm, cf MD_AR_MIRROR_TOP
+    :param focus_dist: m, cf MD_AR_FOCUS_DISTANCE
+    :param x_max: m, cf MD_AR_XMAX
+    :param parabola_f: m, cf MD_AR_PARABOLA_F
+    :param length > 0: the number of angles returned
+    :returns:
```
Comment (K4rishma) [inline src/odemis/util/angleres.py:681]: same here

File: src/odemis/util/test/angleres_test.py [hunk 3]
```diff
@@ -441,10 +440,85 @@ def test_uint16_input_rect2polar(self):
...
+        numpy.testing.assert_array_almost_equal(da_lin, 0)
+
+
 if __name__ == "__main__":
     # for debug:
     # import sys;sys.argv = ['', 'TestPolarConversionOutput.test_2000x2000']
     unittest.main()
     # suite = unittest.TestLoader().loadTestsFromTestCase(TestPolarConversionOutput)
     # unittest.TextTestRunner(verbosity=2).run(suite)
-
```
Comment (K4rishma) [inline src/odemis/util/test/angleres_test.py:450]: Should we omit an extra line at the end?

**Review by K4rishma — APPROVED** [review]

File: src/odemis/util/angleres.py (hunk 2 above)
Comment (pieleric) [reply src/odemis/util/angleres.py:681]: Every function in this file is documented with `returns`. I guess it's automatically completed like that by PyCharm. I'm not going to update them. Just the functions I've touched.

File: src/odemis/util/test/angleres_test.py (hunk 3 above)
Comment (pieleric) [reply src/odemis/util/test/angleres_test.py:450]: Yes, I think so. I think the rule is that the last line should have a `\n` at the end. This file had two `\n` at the end, showing up as one empty line. No need to have an empty line. Github will show a special read icon if the file is missing `\n` at the end, which is not the case here.

**Pull request merged**